rust/connlib/bufferpool/lib.rs

```diff
@@ -25,26 +40,28 @@ where
             .build();
 
         Self {
-            inner: Arc::new(lockfree_object_pool::SpinLockObjectPool::new(
-                move || {
-                    BufferStorage::new(
-                        B::with_capacity(capacity),
-                        buffer_counter.clone(),
-                        [
-                            KeyValue::new("system.buffer.pool.name", tag),
-                            KeyValue::new("system.buffer.pool.buffer_size", capacity as i64),
-                        ],
-                    )
-                },
-                |_| {},
-            )),
+            inner: Arc::new(SegQueue::new()),
+
+            // TODO: It would be nice to eventually create a fixed amount of buffers upfront.
```

Comment:
Just for my own education, this would prevent having to allocate later on, correct? We'd essentially allocate all we need for packet processing up front.

Reply:
Yeah, it is just an idea at this point. That would mean predictable memory usage but we also need to precisely balance, how many buffers we have (IP packet vs UDP packet).